ascot-axum/examples/fridge/src/main.rs

```diff
@@ -0,0 +1,115 @@
+extern crate alloc;
+
+mod dummy_fridge;
+
+use alloc::sync::Arc;
+
+use ascot_axum::devices::fridge::Fridge;
+use async_lock::Mutex;
+use serde::Serialize;
+
+// Library.
+use ascot_library::hazards::Hazard;
+use ascot_library::input::Input;
+use ascot_library::route::Route;
+
+// Miscellaneous.
+use ascot_axum::error::Error;
+use ascot_axum::service::ServiceBuilder;
+
+// Device.
+use ascot_axum::extract::{Extension, Path};
+
+use ascot_axum::device::{DeviceAction, DeviceError, DevicePayload};
+
+// Server.
+use ascot_axum::server::AscotServer;
+
+// Fake fridge implementation
+use dummy_fridge::DummyFridge;
+
+#[derive(Clone, Default)]
+struct DeviceState(Arc<Mutex<DummyFridge>>);
+
+impl DeviceState {
+    fn new(fridge: DummyFridge) -> Self {
+        Self(Arc::new(Mutex::new(fridge)))
+    }
+}
+
+impl core::ops::Deref for DeviceState {
+    type Target = Arc<Mutex<DummyFridge>>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl core::ops::DerefMut for DeviceState {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.0
+    }
+}
+
+#[derive(Serialize)]
+struct ChangeTempResponse {
+    temperature: f64,
+}
+
+async fn increase_temperature(
+    Path(increment): Path<f64>,
+    Extension(state): Extension<DeviceState>,
+) -> Result<DevicePayload, DeviceError> {
+    let mut fridge = state.lock().await;
+    fridge.increase_temperature(increment);
+
+    DevicePayload::new(ChangeTempResponse {
+        temperature: fridge.temperature,
+    })
+}
+
+async fn decrease_temperature(
+    Path(decrement): Path<f64>,
+    Extension(state): Extension<DeviceState>,
+) -> Result<DevicePayload, DeviceError> {
+    let mut fridge = state.lock().await;
+    fridge.decrease_temperature(decrement);
+
+    DevicePayload::new(ChangeTempResponse {
+        temperature: fridge.temperature,
+    })
+}
+
```

Comment:
Can we add a `POST` API as we have done for `light` too? In this way we can check a `POST` route too.